src/mon/MgrStatMonitor.cc

```diff
@@ -69,6 +103,29 @@ void MgrStatMonitor::create_initial()
 void MgrStatMonitor::calc_pool_availability()
 {
   dout(20) << __func__ << dendl;
+  std::scoped_lock l(lock);
+
+  // if feature is disabled by user, do not update the uptime 
+  // and downtime, exit early
+  if (!enable_availability_tracking) {
+    dout(20) << __func__ << "tracking availability score is disabled" << dendl;
+    return;
+  }
+
+  // if reset_availability_last_uptime_downtime_val is not utime_t(1, 2), 
+  // update last_uptime and last_downtime for all pools to the 
+  // recorded values
+  if (reset_availability_last_uptime_downtime_val != utime_t(1, 2)) {
+    for (const auto& i : pool_availability) {
+      const auto& poolid = i.first;
+      pool_availability[poolid].last_downtime = reset_availability_last_uptime_downtime_val;
+      pool_availability[poolid].last_uptime = reset_availability_last_uptime_downtime_val;
+    }
+    dout(20) << __func__ << " reset last_uptime and last_downtime to " 
+             << reset_availability_last_uptime_downtime_val << dendl;
+    reset_availability_last_uptime_downtime_val = utime_t(1, 2);
+  }
+
```

Comment:
Nice! hopefully it solves the `update_paxos` decoding new values problem we discussed on Wednesday!

Reply:
Yes, I tested it out and it seems to be working all fine!